### ISSUE TITLE
daemonset pod scheduled but always in pending state

### ISSUE THREAD

**Issue by aagu** [kubeedge/kubeedge#3019]
**What happened**:
In my cluster, I deploy the edgemesh using daemonset along with cloudcore. When edge node was added to the cluster, the daemonset create a pod and k8s scheduled it to the edge node, but this pod was always in Pending state, unless manually delete the pod to force a pod schedule to the edge node, so the new pod can run on this edge node.
**What you expected to happen**:
The first schedule pod should running on edge node.
**How to reproduce it (as minimally and precisely as possible)**:
1. deploy cloudcore using deployment
2. deploy edgemesh using daemonset
3. add edge node and disable built-in edgemesh
4. the edgemesh deamonset schedules a pod to edge node, but remain in Pending state
**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.17.2
- KubeEdge version(e.g. `cloudcore --version` and `edgecore --version`): v1.7.1
- <details><summary>Cloud nodes Environment:</summary>

  - Hardware configuration (e.g. `lscpu`):
  - OS (e.g. `cat /etc/os-release`):
  - Kernel (e.g. `uname -a`):
  - Go version (e.g. `go version`):
  - Others:

  </details>
- <details><summary>Edge nodes Environment:</summary>

  - edgecore version (e.g. `edgecore --version`): v1.7.1
  - Hardware configuration (e.g. `lscpu`): 4 cores
  - OS (e.g. `cat /etc/os-release`): CentOS 7
  - Kernel (e.g. `uname -a`): Linux 3.10.0-1127.el7.x86_64 #1 SMP Tue Mar 31 23:36:51 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
  - Go version (e.g. `go version`):
  - Others:

  </details>

**Comment by ZBoIsHere** [comment]
@aagu hello, could you apply another daemonset like hostname to see if every node has a helathy hostname pod

**Comment by aagu** [comment]
@ZBoIsHere Hi, as you suggested, I applied another daemonset like this:
```
apiVersion: apps/v1
kind: DaemonSet
metadata:
  name: hostname
  namespace: kubeedge
spec:
  selector:
    matchLabels:
      app: hostname
  template:
    metadata:
      labels:
        app: hostname
    spec:
      containers:
        - name: hostname
          image: 'registry.cn-hangzhou.aliyuncs.com/ks8gcr/serve_hostname:latest'
          resources:
            limits:
              cpu: 230m
              memory: 200Mi
            requests:
              cpu: 220m
              memory: 200Mi
          imagePullPolicy: Always
      restartPolicy: Always
      hostNetwork: true
      securityContext: {}
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
              - matchExpressions:
                  - key: node-role.kubernetes.io/edge
                    operator: Exists
                  - key: node-role.kubernetes.io/agent
                    operator: Exists
      schedulerName: default-scheduler
      tolerations:
        - operator: Exists
```
but the status of this pod scheduled to edge node remain in Pending.

After dug into edgecore logs, I found this error message: 
```
der:{ID:1f6e3606-e951-41fd-b2c0-69ec1fe36765 ParentID: Timestamp:1628823714944 ResourceVersion: Sync:true} Router:{Source:edged Group:meta Operation:query Resource:kubeedge/secret/default-token-675jn} Content:<nil>}], resp[{Header:{ID: ParentID: Timestamp:0 ResourceVersion: Sync:false} Router:{Source: Group: Operation: Resource:} Content:<nil>}], err[timeout to get response]
Aug 13 11:03:54 gq-node1 edgecore[12709]: E0813 11:03:54.955767   12709 process.go:471] remote query failed: timeout to get response
Aug 13 11:03
```

However, a pod from another namespace controlled by deployment successfully running on this node:
![image](https://user-images.githubusercontent.com/7698210/129299723-9199a7cd-0877-4b5f-85ba-b0468934e472.png)

I don't think this was caused by network failure, since I've delete the pod scheduled to edge node several times but the problem remians.

**Comment by aagu** [comment]
I've also checked that meteserver worked as intented:
```
[root@gq-node1 ~]# curl http://127.0.0.1:10550/api/v1/namespaces/kubeedge/secrets/default-token-675jn
{"apiVersion":"v1","data":{"ca.crt":"some data here"},"kind":"Secret","metadata":{"annotations":{"kubernetes.io/service-account.name":"default","kubernetes.io/service-account.uid":"cc0d5880-c080-4c28-9218-f50732cc472a"},"creationTimestamp":"2021-08-10T10:11:00Z","name":"default-token-675jn","namespace":"kubeedge","resourceVersion":"9555625","selfLink":"kubeedge/secrets/default-token-675jn","uid":"e47aa334-43a3-4ddb-a642-5355315d8a1d"},"type":"kubernetes.io/service-account-token"}[root@gq-node1 ~]#
```
I don't known if metesever takes controll over meteclient in edged in kubeedge v1.7.1?

**Comment by aagu** [comment]
@ZBoIsHere hello, is here any progress of this issue?

**Comment by stingshen** [comment]
This may be related. https://github.com/kubeedge/kubeedge/issues/3108#issuecomment-969943066

In conclusion, the issue should be resolved on v1.9.0.
